File: backend/app/story_audio.py
from dataclasses import dataclass
from typing import Protocol


@dataclass(frozen=True)
class StoryAudio:
    audio_status: str = "unavailable"
    audio_url: str | None = None
    audio_voice: str | None = None
    audio_duration: int | None = None


class TtsProvider(Protocol):
    def generate(self, *, story_id: str, text: str, voice: str | None = None) -> StoryAudio: ...


class AudioGenerationService(Protocol):
    def enqueue(self, story_id: str) -> None: ...

    def regenerate(self, story_id: str) -> None: ...

File: lib/services/audio/story_audio.dart
class StoryAudio {
  const StoryAudio({
    this.audioStatus = 'unavailable',
    this.audioUrl,
    this.audioVoice,
    this.audioDuration,
  });

  final String audioStatus;
  final String? audioUrl;
  final String? audioVoice;
  final int? audioDuration;
}

abstract interface class TtsProvider {
  Future<StoryAudio> generate({
    required String storyId,
    required String text,
    String? voice,
  });
}

abstract interface class AudioGenerationService {
  Future<void> enqueue(String storyId);

  Future<void> regenerate(String storyId);
}
